### PR TITLE
Remove -x flag for bonsai CLI options

### PR DESCRIPTION
**Pull request by frankisawesome**
Signed-off-by: Frank Li <b439988l@gmail.com>

## PR description
Removes -X from the bonsai flags so users feel safer using the options as bonsai have been tested extensively, note forest will still be the default storage format.

## Changelog
Add to breaking changes (should we support the -X flag for a little longer?)

- [x] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

**Review by matkt — APPROVED** [review]
LGTM

**Review by shemnon — APPROVED** [review]
This is fine.  The `--X` part of the option means that there is no commitment to keep it working or the same between any two releases. Removing it means changing it in a backwards incompatible way or removing it only occurs at quarterly releases.

**Review by diega — COMMENTED** [review]
Are these options still considered _unstable_? if not, I think we need to change the package and the `unstableDataStorageOptions` field of the `BesuCommand`

**Comment by matkt** [comment]
> Are these options still considered _unstable_? if not, I think we need to change the package and the `unstableDataStorageOptions` field of the `BesuCommand`

no, it's not unstable. good catch we can change that

File: besu/src/main/java/org/hyperledger/besu/cli/options/unstable/DataStorageOptions.java [hunk 1]
```diff
@@ -29,10 +29,10 @@
 
 public class DataStorageOptions implements CLIOptions<DataStorageConfiguration> {
 
-  private static final String DATA_STORAGE_FORMAT = "--Xdata-storage-format";
+  private static final String DATA_STORAGE_FORMAT = "--data-storage-format";
```
Comment (macfarla) [inline besu/src/main/java/org/hyperledger/besu/cli/options/unstable/DataStorageOptions.java:32]: as per comments, can the package name here be changed to remove "unstable"

**Review by macfarla — CHANGES_REQUESTED** [review]
just need to remove "unstable" from the package name for DataStorageOptions

File: besu/src/main/java/org/hyperledger/besu/cli/options/unstable/DataStorageOptions.java (hunk 1 above)
Comment (frankisawesome) [reply besu/src/main/java/org/hyperledger/besu/cli/options/unstable/DataStorageOptions.java:32]: [8bd3d10](https://github.com/hyperledger/besu/pull/3578/commits/8bd3d10e473dc7fc2984c924161b484ebadd1e48)

**Review by macfarla — APPROVED** [review]

**Pull request merged**